tests/chainer_tests/link_hooks_tests/test_weight_standardization.py

```diff
@@ -0,0 +1,190 @@
+
+    def _init_layer(self):
+        hook = WeightStandardization()
+        layer = self.layer
+        layer.add_hook(hook)
+        if self.lazy_init:
+            # Initialize weight and bias.
+            with chainer.using_config('train', False):
+                layer(self.x)
+        return layer, hook
+
+    def check_weight_is_parameter(self, gpu):
+        layer, hook = self._init_layer()
+        if gpu:
+            layer = layer.to_gpu()
+        source_weight = getattr(layer, hook.weight_name)
+        x = cuda.to_gpu(self.x) if gpu else self.x
+        layer(x)
+        assert getattr(layer, hook.weight_name) is source_weight
+
+    def test_weight_is_parameter_cpu(self):
+        if not self.lazy_init:
+            self.check_weight_is_parameter(False)
+
+    @attr.gpu
+    def test_weight_is_parameter_gpu(self):
+        if not self.lazy_init:
+            self.check_weight_is_parameter(True)
+
+    def check_deleted(self, gpu):
+        layer, hook = self.layer, self.hook
+        layer.add_hook(hook)
+        if gpu:
+            layer = layer.to_gpu()
+        x = cuda.to_gpu(self.x) if gpu else self.x
+
+        y1 = layer(x).array
+        with chainer.using_config('train', False):
+            y2 = layer(x).array
+        layer.delete_hook(hook.name)
+        y3 = layer(x).array
+        if gpu:
+            y1, y2, y3 = cuda.to_cpu(y1), cuda.to_cpu(y2), cuda.to_cpu(y3)
+        assert not numpy.array_equal(y1, y3)
+        assert not numpy.array_equal(y2, y3)
+
+    def test_deleted_cpu(self):
+        self.check_deleted(False)
+
+    @attr.gpu
+    def test_deleted_gpu(self):
+        self.check_deleted(True)
+
+
+class TestEmbedID(unittest.TestCase, BaseTest):
+
+    def setUp(self):
+        self.lazy_init = False  # For convenience.
+        self.bs, self.in_size, self.out_size = 5, 10, 20
+        self.x = numpy.arange(self.in_size, dtype=numpy.int32)
+        self.layer = L.EmbedID(self.in_size, self.out_size)
+        self.hook = WeightStandardization()
+
+    def test_add_ws_hook(self):
+        hook = WeightStandardization()
+        layer = self.layer
+        layer.add_hook(hook)
+        if self.lazy_init:
+            with chainer.using_config('train', False):
+                layer(self.x)
+
+
+@testing.parameterize(*testing.product({
+    'lazy_init': [True, False],
+}))
+class TestLinear(unittest.TestCase, BaseTest):
+
+    def setUp(self):
+        self.bs, self.in_size, self.out_size = 10, 20, 30
+        self.x = numpy.random.normal(
+            size=(self.bs, self.in_size)).astype(numpy.float32)
+        self.layer = L.Linear(self.out_size)  # Lazy initialization
+        in_size = None if self.lazy_init else self.in_size
+        self.layer = L.Linear(in_size, self.out_size)
+        self.hook = WeightStandardization()
+
+
+@testing.parameterize(*testing.product({
+    'lazy_init': [True, False],
+    'link': [L.Convolution1D, L.Deconvolution1D],
```

Comment:
Do we need a test for Deconvolution layer?

Reply:
Not be needed, removed it.